### PR TITLE
Added @warn_unused_result to functions declarations

### PR DESCRIPTION
**Pull request by Austinate**
PR for #18

### CODE REVIEW

**Comment by Austinate** [comment]
Those trailing whitespaces 😕

File: Mapper/Mappable.swift [hunk 1]
```diff
@@ -33,10 +34,13 @@ public protocol Mappable {
 
      - returns: The object if it could be created, nil if creating the object threw an error
      */
+    @warn_unused_result
     static func from(JSON: NSDictionary) -> Self?
 }
 
 public extension Mappable {
+
```
Comment (keith) [inline Mapper/Mappable.swift]: Extra newline!

**Comment by keith** [comment]
Thanks for doing this! Do you mind adding it as an enhancement in the CHANGELOG?

File: Mapper/Mapper.swift [hunk 2]
```diff
@@ -342,6 +361,7 @@ public struct Mapper {
 
      - returns: The object for the given key
      */
+    @warn_unused_result
     private func JSONFromField(field: String) -> AnyObject? {
```
Comment (Reflejo) [inline Mapper/Mapper.swift]: Wouldn't add it to private functions

File: Mapper/Errors.swift [hunk 3]
```diff
@@ -5,5 +5,6 @@
  Custom implementations of Mappable, Convertible, or transformation functions can throw any error they desire
  */
 public struct MapperError: ErrorType {
+    @warn_unused_result
```
Comment (keith) [inline Mapper/Errors.swift]: Hmmm. Looks like this warning actually happens without `@warn_unused_result` as well. With slightly different text. Lets leave this here anyways to be explicit though.

File: Mapper/Mapper.swift (hunk 2 above)
Comment (keith) [inline Mapper/Mapper.swift]: I think we should leave it on this one? Although it should be obvious we don't want people to use this incorrectly in the Mapper implementation either?

Comment (Reflejo) [inline Mapper/Mapper.swift]: My reasoning is that we have control on what happens `private`ly and this just seems overprotective. Is it realistic to think that `JSONFromField` will be misused?.

Comment (keith) [inline Mapper/Mapper.swift]: Fair, sounds good, let's remove this one @Austinate

**Comment by keith** [comment]
A few tiny fixes and the changelog entry and this will be good to merge! Please squash after those fixes too!

**Comment by Austinate** [comment]
Should be good to go now

**Comment by keith** [comment]
Awesome! Merged here: a8f23ad3e4e806dae71be1f266285eb7f3ea2c19

**Comment by keith** [comment]
Thanks!

**Comment by Austinate** [comment]
Welcome :)

**Pull request closed without merging**